lib/simulator-xcode-6.js

```diff
@@ -272,59 +273,60 @@ class SimulatorXcode6 extends EventEmitter {
     return indicator;
   }
 
-  async run (opts = {}) {
-    if (await this.isRunning() && await this.isUIClientRunning()) {
-      log.info(`Simulator with udid '${this.udid}' already booted.`);
-      return;
-    }
-    log.info(`Simulator with udid '${this.udid}' not booted. Booting up now`);
-    try {
-      // Make sure Simulator is in Shutdown state before booting it
-      await this.shutdown();
-    } catch (ign) {
-      // ignore error
-    }
+  async startUIClient (opts = {}) {
     opts = Object.assign({
       scaleFactor: null,
       connectHardwareKeyboard: false,
+      startupTimeout: this.startupTimeout,
+    }, opts);
+
+    let simulatorApp = path.resolve(await getXcodePath(), 'Applications', this.simulatorApp);
+    let args = ['-Fn', simulatorApp, '--args', '-CurrentDeviceUDID', this.udid];
+
+    if (opts.scaleFactor) {
+      const supportedScales = ['1.0', '0.75', '0.5', '0.33', '0.25'];
+      if (supportedScales.indexOf(opts.scaleFactor) < 0) {
+        log.errorAndThrow(`Only "${supportedScales}" values are supported as scale factors. "${opts.scaleFactor}" is passed instead.`);
+      }
+      const stat = await this.stat();
+      const formattedDeviceName = stat.name.replace(/\s+/g, '-');
+      const argumentName = `-SimulatorWindowLastScale-com.apple.CoreSimulator.SimDeviceType.${formattedDeviceName}`;
+      args.push(argumentName, opts.scaleFactor);
+    }
+
+    if (!opts.connectHardwareKeyboard) {
+      args.push('-ConnectHardwareKeyboard', '0');
+    }
+
+    log.info(`Starting Simulator UI with command: open ${args.join(' ')}`);
+    await exec('open', args, {timeout: opts.startupTimeout});
+  }
+
+  async run (opts = {}) {
+    opts = Object.assign({
       allowTouchEnroll: false,
       startupTimeout: this.startupTimeout,
     }, opts);
+    const {state} = await this.stat();
+    const isServerRunning = state === 'Booted';
+    const isUIClientRunning = await this.isUIClientRunning();
+    if (isServerRunning && isUIClientRunning) {
+      log.info('Both Simulator with UDID ${this.udid} and the UI client are currently running. There\'s no need for restart.');
```

Comment:
This needs to be in backtick quotes to get the template insertion. Then the escape of the apostrophe is also unneeded.